src/plugins/cloudMetadata.nim

```diff
@@ -150,7 +156,35 @@ proc isAwsEc2Host(vendor: string): bool =
   return false
 
 proc isGoogleHost(vendor: string): bool =
-  return contains(strutils.toLowerAscii(vendor), "google")
+  # vendor is present
+  if contains(strutils.toLowerAscii(vendor), "google"):
+    return true
+
+  # vendor information should be present in most services, but its not present
+  # in cloud run. In cloud run we can detect the presence of a knative service
+  # via ENV variables but we are being conservative in also checking resolv.conf
+  let resolvContents = tryToLoadFile("/etc/resolv.conf")
+  var hasGoogleInternal = false
+  for line in resolvContents.splitLines():
+    # Checking that resolv.conf contains `google.internal` outside of a comment
+    # should be more than sufficient.
+    #
+    # From `man resolv.conf`:
+    #
+    # - The keyword and value must appear on a single line, and the keyword
+    #   (e.g., nameserver) must start the line.  The value follows the keyword,
+    #   separated by white space.
+    #
+    # - Lines that contain a semicolon (;) or hash character (#) in the first
+    #   column are treated as comments.
+    if line.len() > 0 and line[0] notin {';', '#'} and line.contains("google.internal"):
+      hasGoogleInternal = true
+      break
+  return (hasGoogleInternal and
+          getEnv(CLOUD_RUN_TIMEOUT_SECONDS) != "" and
+          getEnv(K_SERVICE) != "")
```

Comment:
how is `K_SERVICE` related to GCP cloud run? are all cloud run containers deployed as knative services? are the docs for this?

not all customers might be using knative in their infra so unless GCP itlself uses knative I dont think we should be tying knative usage with any cloud deployment

Reply:
yes its deployed as a knative service